app/pages/lab/workflow-viewer/task.cjsx

```diff
@@ -0,0 +1,313 @@
+        else
+          L = Object.keys(@state.task.tools).length
+        for idx in [0...L]
+          id = @getUuid(idx, uuid, uuids)
+          uuids.push(id)
+          uuid +=1
+          switch @state.task.type
+            when 'single' then ep = @props.jp.addEndpoint(id, Sty.commonA, {uuid: id})
+            when 'drawing' then ep = @props.jp.addEndpoint(id, Sty.commonA_open, {uuid: id})
+          ep.canvas.style['z-index'] = @state.zIndex
+          eps.push(ep)
+        @setUuid(null, uuid, uuids)
+      if @state.task.type != 'single'
+        ep = @props.jp.addEndpoint(@props.plumbId+'_name', Sty.commonA, {uuid: @props.plumbId + '_next'})
+        ep.canvas.style['z-index'] = @state.zIndex
+        eps.push(ep)
+    computedStyle = window.getComputedStyle(me)
+    new_state =
+      endpoints: eps
+      me: me
+      width:
+        min: parseFloat(computedStyle['min-width'])
+        max: parseFloat(computedStyle['max-width'])
+        val: parseFloat(computedStyle['width'])
+    @setState(new_state, @props.onDone)
+
+  # when the node is removed clean up all endpoints
+  componentWillUnmount: ->
+    for ep in @state.endpoints
+      @props.jp.deleteEndpoint(ep)
+
+  # define some setters and getters
+  #==========================
+
+  # Keep track of unique uuids to use for each answer DOM element
+  # These can't be re-used after removal, otherwise the endpoint will not draw in the right spot
+  # This is a limitation of jsPlumb
+  setUuid: (id, uuid, uuids) ->
+    current_uuids = uuids ? @state.uuids.concat([id])
+    current_uuid =  uuid ? @state.uuid + 1
+    @setState({uuids: current_uuids, uuid: current_uuid})
+
+  # Get a unique uuid for an answer based on its position in the answer list
+  getUuid: (idx, uuid = @state.uuid, uuids = @state.uuids) ->
+    # When loading with answsers make sure to use idx since setState is not called until after the loop
+    if uuids[idx]?
+      return @state.uuids[idx]
+    else
+      return @props.plumbId + '_answer_' + idx
+
+  # add an endpoint to the list of endpoints
+  pushEps: (ep) ->
+    eps = @state.endpoints.concat([ep])
+    @setState({endpoints: eps})
+
+  # remove an endpoint from the list of endpoints (always remove the last one)
+  removeEps: ->
+    eps = @state.endpoints[...-1]
+    current_uuids = @state.uuids[...-1]
+    @setState({endpoints: eps})
+    @setState({uuids: current_uuids})
+
+  # get css style to use
+  getStyle: ->
+    #style = @props.pos
+    if @state.me?
+      style =
+        left: @state.me.style.left
+        top: @state.me.style.top
+        width: @state.me.style.width
+        zIndex: @state.zIndex
+      if parseInt(style.width) > @state.width.max
+        style.width = @state.width.max + 'px'
+      if parseInt(style.width) < @state.width.min
+        style.width = @state.width.min + 'px'
+    else
+      style = @props.pos
+      style['zIndex'] = @state.zIndex
+    if @state.zIndex > max_z
+      max_z = @state.zIndex
+    style
```

Comment:
If you need to track left, top, width, etc as state, set them as state rather than saving them indirectly on the style object of the component.

Pseudo-code below, not tested but might look like this:

```
getStyle: ->
  width = if @state.width > @state.maxWidth
    @state.maxWidth
  else if @state.width < @state.minWidth
    @state.minWidth
  else
    @state.width

  if @state.zIndex > currentMaxZIndex
    currentMaxZIndex = @state.zIndex

  left: @state.left
  top: @state.top
  width: width
  zIndex: @state.zIndex
  ```